### PR TITLE
CSHARP-4492: Drivers should not create the ECC collection in v2 of queryable encryption.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Operations/CreateCollectionOperation.cs [hunk 1]
```diff
@@ -82,6 +81,7 @@ CreateCollectionOperation CreateInnerCollectionOperation(string collectionName)
         private readonly MessageEncoderSettings _messageEncoderSettings;
         private bool? _noPadding;
         private BsonDocument _storageEngine;
+
```
Comment (JamesKovacs) [inline src/MongoDB.Driver.Core/Core/Operations/CreateCollectionOperation.cs:84]: Extraneous newline.

File: src/MongoDB.Driver/Encryption/ClientEncryption.cs [hunk 2]
```diff
@@ -119,6 +120,7 @@ public CreateEncryptedCollectionResult CreateEncryptedCollection(IMongoDatabase
             Ensure.IsNotNull(collectionName, nameof(collectionName));
             Ensure.IsNotNull(createCollectionOptions, nameof(createCollectionOptions));
             Ensure.IsNotNull(kmsProvider, nameof(kmsProvider));
+            EnsureFeatureSupported(database.Client.Cluster, Feature.Csfle2QEv2, cancellationToken);
```
Comment (JamesKovacs) [inline src/MongoDB.Driver/Encryption/ClientEncryption.cs:123]: Won't this prevent FLEv1 from being used with newer drivers?

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Two issues...
- Question about whether this change prevents new drivers from using FLEv1 features.
- Minor formatting issue.

File: src/MongoDB.Driver.Core/Core/Operations/CreateCollectionOperation.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/Core/Operations/CreateCollectionOperation.cs:84]: done

File: src/MongoDB.Driver/Encryption/ClientEncryption.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Driver/Encryption/ClientEncryption.cs:123]: Inside this method there is a validation on encryptedFields value which is part of CSFLE2. So, FLE1 should be not affected. NOTE: I'm not sure off the top of my head whether it's even possible to be in this mode with latest libmongocrypt bingins

File: tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs [hunk 3]
```diff
@@ -172,6 +172,42 @@ private void AssertCommandAspect(BsonDocument actualCommand, string name, BsonVa
...
+                        if (commandName == "create") // create encrypted collection
+                        {
+                            var encryptedFieldsDocument = actualCommand[name].AsBsonDocument;
+                            foreach (var expectedItem in expectedValue.AsBsonDocument)
+                            {
+                                switch (expectedItem.Name)
+                                {
+                                    case "escCollection":
+                                    case "ecocCollection" :
+                                    case "eccCollection":
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs:185]: `eccCollection` was removed. We shouldn't need this case anymore.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Question about the asserter and then we're good.

File: tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs (hunk 3 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs:185]: in theory yes, but json tests assert that this value is not filled

Comment (JamesKovacs) [reply tests/MongoDB.Driver.Core.TestHelpers/JsonDrivenTests/CommandStartedEventAsserter.cs:185]: Ah ok. LGTM then.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**